gpflow/conditionals/util.py

```diff
@@ -103,7 +118,15 @@ def base_conditional(
     if not full_cov:
         fvar = tf.linalg.adjoint(fvar)  # [N, R]
 
-    return fmean, fvar  # [N, R], [R, N, N] or [N, R]
+    shape_constraints = [
+        (Kmn, [..., "M", "N"]),  # tensor included again for N dimension
+        (f, [..., "M", "R"]),  # tensor included again for R dimension
```

Comment:
what do these comments mean?

Reply:
that I'm including this tensor in the list so I get a handle on the `N` and `R` respectively.... I tried to improve it following previous review comments!